Cannot include immintin.h on LLVM >= 10

Submitting this an issue now so that I can keep track of a test failure in LLVM 10 and greater.

```
./terra tests/expvec.t
```

results in a lot of spew like:

```
/usr/include/clang/11.1.0/include/avx512vlbwintrin.h:1897:46: error: passing '__v8hi' (aka 'short') to parameter of incompatible type '__att
ribute__((__vector_size__(8 * sizeof(short)))) short' (vector of 8 'short' values)
                                             (__v8hi)_mm_sllv_epi16(__A, __B),
                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I'm turning the test off for now so I can move forward with LLVM 10-11.

--- a/release/share/terra/README.md
+++ b/release/share/terra/README.md
@@ -145,8 +145,8 @@ The current recommended version of LLVM is **6.0**. The following versions are a
 | 7.0 | :heavy_check_mark: | :heavy_check_mark: | requires CMake |
 | 8.0 | :heavy_check_mark: | :heavy_check_mark: | |
 | 9.0 | :heavy_check_mark: | :heavy_check_mark: | |
-| 10.0 | :heavy_check_mark: | :heavy_check_mark: | [#476](https://github.com/terralang/terra/issues/476) [#477](https://github.com/terralang/terra/issues/477) |
-| 11.0 | :heavy_check_mark: | :heavy_check_mark: | [#476](https://github.com/terralang/terra/issues/476) [#477](https://github.com/terralang/terra/issues/477) |
+| 10.0 | :heavy_check_mark: | :heavy_check_mark: | [#477](https://github.com/terralang/terra/issues/477) |
+| 11.0 | :heavy_check_mark: | :heavy_check_mark: | [#477](https://github.com/terralang/terra/issues/477) |
 
 ### Instructions for Building LLVM from Source
 

--- a/src/terralib.lua
+++ b/src/terralib.lua
@@ -3432,9 +3432,9 @@ function terra.includecstring(code,cargs,target)
       args:insert("-isysroot")
       args:insert(sdkroot)
     end
-    -- Set GNU C version to match values observed on macOS.
-    if ffi.os == "OSX" and terralib.llvm_version >= 100 then
-      args:insert("-fgnuc-version=4")
+    -- Set GNU C version to match value set by Clang: https://github.com/llvm/llvm-project/blob/f77c948d56b09b839262e258af5c6ad701e5b168/clang/lib/Driver/ToolChains/Clang.cpp#L5750-L5753
+    if ffi.os ~= "Windows" terralib.llvm_version >= 100 then
+      args:insert("-fgnuc-version=4.2.1")
     end
 
     if cargs then

--- a/src/terralib.lua
+++ b/src/terralib.lua
@@ -3433,7 +3433,7 @@ function terra.includecstring(code,cargs,target)
       args:insert(sdkroot)
     end
     -- Set GNU C version to match value set by Clang: https://github.com/llvm/llvm-project/blob/f77c948d56b09b839262e258af5c6ad701e5b168/clang/lib/Driver/ToolChains/Clang.cpp#L5750-L5753
-    if ffi.os ~= "Windows" terralib.llvm_version >= 100 then
+    if ffi.os ~= "Windows" and terralib.llvm_version >= 100 then
       args:insert("-fgnuc-version=4.2.1")
     end
 

--- a/tests/expvec.t
+++ b/tests/expvec.t
@@ -3,10 +3,6 @@ if require("ffi").os == "Windows" then
 	print("Not consistent on windows")
 	return
 end
-if terralib.llvm_version >= 100 then
-	print("FIXME #476 cannot include immintrin.h on LLVM >= 10")
-	return
-end
 local C = terralib.includecstring[[
 	#include <immintrin.h>
 	void dostuff(__m128 what) {}

